### DISCUSSION TITLE
How to intercept image pasting?

### DISCUSSION THREAD

**Discussion by gustavotoyota** [ueberdosis/tiptap#2988] [General Questions]
I need to have only base64 images in the editor. I already developed a form to insert base64 images. The problem is that the user can still paste URL images into the editor. How can I intercept image pasting in order to replace the pasted images with base64?

Thank you.

**Comment by bigsee** [comment]
You might be want to create a custom ProseMirror plugin inside your image extension where you can then override the internal paste handler method with your own implementation. So something like this might work inside your extension:

``` js
addProseMirrorPlugins() {
    return [
      new Plugin({
        props: {
          handlePaste: (view, event) => {

            // your logic here
            
            if (userBehaviourWeWantToBlockHappens || payloadDoesNotContainAnImage) {
                return false
            }

            if (userPastedImageByURL) {
                convertImageToBase64()
            }

            return true
          }
        }
      })
    ]
  }
```
